_drafts/2017-07-05-what-is-a-headless-cms.md

```diff
@@ -0,0 +1,100 @@
+ Thus, the 'E' in ECM is for Enterprise.
+
+  [ed1adc41]: https://www.sensenet.com/product "Sensenet ECM Product features"
+  [a19fba28]: http://wiki.sensenet.com/Versioning_and_approval "Versioning and approval"
+  [206457f6]: http://wiki.sensenet.com/Workspace "Workspaces in sensenet ECM"
+
+This is why you could do something like this in [sensenet 6.x][9f950a61], or using [sn-webpages 7.x][46ac6b1f]:
+```html
+<div class="sn-layout-head-left">
+  <div class="sn-logo">
+    <a href="/" title='<%=HttpContext.GetGlobalResourceObject("Portal", "SenseNetDemoSiteTitle")%>' class="snPortalengine">
+    </a>
+  </div>      
+</div>
+```
+<p align="center">
+*fig 1. ~ The template is tightly coupled with the data it needs to present.*
+</p>
+
+  [46ac6b1f]: https://github.com/SenseNet/sn-webpages "Sn-webpages"
+  [9f950a61]: http://community.sensenet.com/docs/how-to-install-sn6/ "Install sensenet 6.5"
+
+In a headless CMS (or ECMS for that matter), though, the backend does not care about how the content is displayed. It just returns with the data needed for the frontend through a RESTful API.
+
+![Headless vs Coupled](../img/posts/headless-cms.png)
+<p align="center">
+*fig 2. ~ Traditional (coupled or monolithic CMS) model vs. Headless CMS*
+</p>
+
+## So is sensenet ECM a headless or a coupled CMS?
+![Why not both?](http://i.imgur.com/KgXtawP.gif)
+
+As mentioned earlier, using sn-webpages, you can have a full-featured ECMS solution, complete with a built-in UI. We are also working on a **new** UI, using [React][b3358601] and [Aurelia][b56bd31b] while fully utilizing [sensenet ECM's OData layer][71579091], which provides a complete, flexible and extendable [RESTful API][f9c00a93].
+
+  [71579091]: http://community.sensenet.com/docs/built-in-odata-actions-and-functions/ "Built-in OData actions and functions"
+  [f9c00a93]: http://community.sensenet.com/docs/odata-rest-api/ "OData REST API"
+  [b3358601]: https://github.com/SenseNet/sn-controls-react "React controls for sensenet ECM"
+  [b56bd31b]: https://github.com/SenseNet/sn-controls-aurelia "Aurelia controls for sensenet ECM"
+
+In the same time, you can either start writing your own library to latch on the OData endpoints of a sensenet ECM at hand, **or** you can try out our [fabulous and open source sn-client-js, the javascript client library for sensenet][de7f7b73].
+
+  [de7f7b73]: https://github.com/SenseNet/sn-client-js "Sn-client-js"
+
+In the end, what you get is something like this <sup>[1](#footnote1)</sup>:
+
+**After installing our library from npm, you can import it** and connect to your sensenet ECM instance.
+```javascript
+import { Repository, Config } from 'sn-client-js';
+
+let repository = new Repository.SnRepository({
+            RepositoryUrl: 'https://my-sensenet-site.com',
+            ODataToken: 'OData.svc',
```

Comment:
All properties are optional in this snippet.  As this post is not focused to authentication or other customization, can we keep only the 'RepositoryUrl' property to keep it simple?